internal/debugagent/debugagent.go

```diff
@@ -0,0 +1,370 @@
+package debugagent
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/agentuity/cli/internal/tools"
+	"github.com/agentuity/go-common/logger"
+	"github.com/anthropics/anthropic-sdk-go"
+	"github.com/charmbracelet/glamour"
+)
+
+// NOTE: I think we should be able to use that fancy go:embed thing here
+// but the import gets nuked when we build the CLI, so doing this nasty
+// init() thing instead.
+var systemPrompt string
+
+func init() {
+	_, file, _, _ := runtime.Caller(0)
+	base := filepath.Dir(file)
+	p := filepath.Join(base, "./debug-system-prompt.txt")
+	if data, err := os.ReadFile(p); err == nil {
+		systemPrompt = string(data)
+	}
+}
+
+type Options struct {
+	Dir           string
+	Error         string
+	Extra         string
+	AllowWrites   bool
+	Logger        logger.Logger
+	MaxIterations int
+}
+
+type Result struct {
+	Analysis string
+	Patch    string // empty if no patch proposed
+	Edited   bool
+}
+
+// cacheEntry stored on disk
+type cacheEntry struct {
+	Analysis string `json:"analysis"`
+}
+
+func Analyze(ctx context.Context, opts Options) (Result, error) {
+	// Helper for conditional trace logging.
+	trace := func(format string, args ...interface{}) {
+		if opts.Logger != nil {
+			opts.Logger.Trace(format, args...)
+		}
+	}
```

Comment:
this is a little weird, we never optionally allow a logger, expect it and the caller can either tune the desired level or pass a `logger.NewNoOpLgger`
```suggestion
	logger := opts.Logger.With(map[string]any{"pkg":"debugagent"})
```